### PR TITLE
Unit test dassert_lt()

### PR DESCRIPTION
**Pull request by jayati1397**
Created this PR for Issue [#1095](https://github.com/kaizen-ai/kaizenflow/issues/1095).

**Changes Made:**
- Implemented unit tests for dassert_lt.

**Testing:**
Ran code through pylint and pytest.
- Pylint Errors: Class name: currently follows the convention specified [here](https://github.com/kaizen-ai/kaizenflow/blob/master/docs/coding/all.write_unit_tests.how_to_guide.md#naming-and-placement-conventions).

```
OUTPUT from the added lines in the file
helpers/test/test_dbg.py:718:0: C0115: Missing class docstring (missing-class-docstring)
helpers/test/test_dbg.py:718:0: C0103: Class name "Test_dassert_lt" doesn't conform to PascalCase naming style (invalid-name)
```
- Pytest: No errors.
```
OUTPUT:
collected 5 items                                                                                                                                       

helpers/test/test_dbg.py::Test_dassert_lt::test1 (0.00 s) PASSED   [ 20%]
helpers/test/test_dbg.py::Test_dassert_lt::test2 (0.00 s) PASSED  [ 40%]
helpers/test/test_dbg.py::Test_dassert_lt::test3 (0.00 s) PASSED  [ 60%]
helpers/test/test_dbg.py::Test_dassert_lt::test4 (0.00 s) PASSED  [ 80%]
helpers/test/test_dbg.py::Test_dassert_lt::test5 (0.00 s) PASSED  [100%]
```

@samarth9008 please let me know if you have any further suggestions

### CODE REVIEW

File: helpers/test/test_dbg.py [hunk 1]
```diff
@@ -713,3 +713,56 @@ def test4(self) -> None:
         Obj = collections.namedtuple("Obj", ["a", "b"])
         list_ = [Obj(1, 2), Obj(1, 2)]
         hdbg.dassert_all_attributes_are_same(list_, "b")
+
+# #############################################################################
```
Comment (samarth9008) [inline helpers/test/test_dbg.py:717]: Leave one line after the separator

File: helpers/test/test_dbg.py [hunk 2]
```diff
@@ -713,3 +713,56 @@ def test4(self) -> None:
...
+class Test_dassert_lt(hunitest.TestCase):
+    
+    def test1(self) -> None:
+        """
+        Test where val1 is less than val2.
```
Comment (samarth9008) [inline helpers/test/test_dbg.py:722]: -> Test that the function works fine if first value is less than second.

The idea is we do not keep variable names in the doc string.

File: helpers/test/test_dbg.py [hunk 3]
```diff
@@ -713,3 +713,56 @@ def test4(self) -> None:
...
+        """
+        hdbg.dassert_lt(1, 2)
```
Comment (samarth9008) [inline helpers/test/test_dbg.py:724]: No magic numbers. Lets do
```
val1 = 1
val2 = 2
hdbg.dassert_lt(val1, val2)
```

make change everywhere.

File: helpers/test/test_dbg.py [hunk 4]
```diff
@@ -713,3 +713,56 @@ def test4(self) -> None:
...
+    
+    def test2(self) -> None:
+        """
+        Test where val1 is equal to val2.
```
Comment (samarth9008) [inline helpers/test/test_dbg.py:728]: Test where val1 is equal to val2. -> Test that the function raises an exception with 2 equal values.

File: helpers/test/test_dbg.py [hunk 5]
```diff
@@ -713,3 +713,56 @@ def test4(self) -> None:
...
+        """
+        with self.assertRaises(AssertionError) as cm:
+            hdbg.dassert_lt(2, 2)
+        act = str(cm.exception)
+        exp = """
+        * Failed assertion *
+        2 < 2
+        """
+        self.assert_equal(act, exp, fuzzy_match=True)
```
Comment (samarth9008) [inline helpers/test/test_dbg.py:737]: Add `# Check.` above this line.

Add everywhere. In your other PRs as well.

File: helpers/test/test_dbg.py [hunk 6]
```diff
@@ -713,3 +713,56 @@ def test4(self) -> None:
...
+    
+    def test3(self) -> None:
+        """
+        Test where val1 is greater than val2.
```
Comment (samarth9008) [inline helpers/test/test_dbg.py:741]:
```
Test that the function raises an exception when first value is greater than second.
```

File: helpers/test/test_dbg.py [hunk 7]
```diff
@@ -713,3 +713,56 @@ def test4(self) -> None:
...
+
+    def test4(self) -> None:
+        """
+        Test where val1 is less than val2 with strings.
+        """
+        hdbg.dassert_lt("a", "b")
+
+    def test5(self) -> None:
+        """
+        Test where val1 is greater than val2 with floats.
```
Comment (samarth9008) [inline helpers/test/test_dbg.py:760]:
```
Test that the function raises an exception when first value is greater than second with float types.
```

File: helpers/test/test_dbg.py [hunk 8]
```diff
@@ -713,3 +713,56 @@ def test4(self) -> None:
...
+        act = str(cm.exception).strip()
+        exp = """
+            * Failed assertion *
+            3 < 2
+        """
+        self.assert_equal(act, exp, fuzzy_match=True)
+
+    def test4(self) -> None:
+        """
+        Test where val1 is less than val2 with strings.
```
Comment (samarth9008) [inline helpers/test/test_dbg.py:754]: Test that the function works with string inputs.

File: helpers/test/test_dbg.py [hunk 9]
```diff
@@ -713,3 +713,56 @@ def test4(self) -> None:
...
+    def test3(self) -> None:
+        """
+        Test where val1 is greater than val2.
+        """
+        with self.assertRaises(AssertionError) as cm:
+            hdbg.dassert_lt(3, 2)
+        act = str(cm.exception).strip()
+        exp = """
+            * Failed assertion *
+            3 < 2
```
Comment (samarth9008) [inline helpers/test/test_dbg.py:748]: indentation could have been better

File: helpers/test/test_dbg.py [hunk 10]
```diff
@@ -713,3 +713,56 @@ def test4(self) -> None:
...
+    def test5(self) -> None:
+        """
+        Test where val1 is greater than val2 with floats.
+        """
+        with self.assertRaises(AssertionError) as cm:
+            hdbg.dassert_lt(2.0, 1.0)
+        act = str(cm.exception)
+        exp = """
+        * Failed assertion *
+        2.0 < 1.0"""
```
Comment (samarth9008) [inline helpers/test/test_dbg.py:767]: Lets bring `"""` to the new line.

**Review by samarth9008 — COMMENTED** [review]
Good PR. Few nits.

File: helpers/test/test_dbg.py (hunk 5 above)
Comment (jayati1397) [reply helpers/test/test_dbg.py:737]: I didn't get this one, can you please elaborate? Shall I just add #check . above assert_equal?

Comment (samarth9008) [reply helpers/test/test_dbg.py:737]: Yes, that is correct. You need to add it everywhere before `self.assert_equal` line in all your PRs.

File: helpers/test/test_dbg.py [hunk 11]
```diff
@@ -713,3 +713,71 @@ def test4(self) -> None:
...
+        """
+        Test that the function works fine if first value is less than second value.
+        """
+        val1 = 1
+        val2 = 2
+        hdbg.dassert_lt(val1, val2)
+    
+    def test2(self) -> None:
+        """
+        Test that the function works fine if first value is equal to second value, this raises an exception.
```
Comment (samarth9008) [inline helpers/test/test_dbg.py:731]: Lets rephrase it a little bit

```
Test that the function raises an exception if first value is equal to second value.
```

Lets make this change everywhere also.

File: helpers/test/test_dbg.py [hunk 12]
```diff
@@ -713,3 +713,71 @@ def test4(self) -> None:
...
+        val1 = 2
+        val2 = 2
+        with self.assertRaises(AssertionError) as cm:
+            hdbg.dassert_lt(val1, val2)
+        act = str(cm.exception)
+        exp = """
+        * Failed assertion *
+        2 < 2
+        """
+        #check.
```
Comment (samarth9008) [inline helpers/test/test_dbg.py:742]: `#check.` -> `# Check.`

Same everywhere.

File: helpers/test/test_dbg.py [hunk 13]
```diff
@@ -713,3 +713,71 @@ def test4(self) -> None:
...
+    
+    def test3(self) -> None:
+        """
+        Test that the function works fine if first value is greater than second value, this raises an exception.
+        """
+        val1 = 3
+        val2 = 2
+        with self.assertRaises(AssertionError) as cm:
+            hdbg.dassert_lt(val1, val2)
+        act = str(cm.exception).strip()
```
Comment (samarth9008) [inline helpers/test/test_dbg.py:753]: Why strip()?

File: helpers/test/test_dbg.py [hunk 14]
```diff
@@ -713,3 +713,71 @@ def test4(self) -> None:
...
+        """
+        Test that the function works with string inputs.
+        """
+        val1 = "a"
+        val2 = "b"
+        hdbg.dassert_lt(val1, val2)
+
+    def test5(self) -> None:
+        """
+        Test where first value is greater than second value with floats, this raises an exception.
```
Comment (samarth9008) [inline helpers/test/test_dbg.py:771]: Test where first value is greater than second value with floats, this raises an exception. -> Test that function raises an exception where first value is greater than second value with float types.

**Review by samarth9008 — COMMENTED** [review]
Final nits.

Sorry if their are repeated corrections. I could make them better with second review.

**Review by gpsaggese — COMMENTED** [review]
LGTM.

A bit paranoid to cover also this basic assertion, but code is well done

**Pull request merged**